core/src/models/Export.ts

```diff
@@ -234,21 +289,52 @@ export class Export extends Model {
         .value
     );
 
-    const _exports = await Export.findAll({
-      where: {
-        mostRecent: false,
-        createdAt: {
-          [Op.lt]: Moment().subtract(days, "days").toDate(),
-        },
-      },
-      order: [["createdAt", "desc"]],
-      limit,
-    });
+    const whereDate = Moment()
+      .subtract(days, "days")
+      .format("YYYY-MM-DD HH:mm:ss");
+
+    const rows: { id: string }[] = await api.sequelize.query(
+      `
+      DELETE FROM exports
+      WHERE id IN (
+        SELECT id FROM exports
+        WHERE "createdAt" < '${whereDate}'
+        AND (
+          "createdAt" < (
+            SELECT MAX("createdAt")
+            FROM exports e2
+            WHERE
+              e2."profileId" = exports."profileId"
+              AND state = 'complete'
+          ) OR (
+            0 = (
+              SELECT COUNT(id)
+              FROM exports e3
+              WHERE
+                e3."profileId" = exports."profileId"
+                AND state = 'complete'
+            )
+          )
+        )
+        LIMIT ${limit}
+      )
+     ${config.sequelize.dialect === "postgres" ? "RETURNING id" : ""}
+      ;`,
+      { type: QueryTypes.SELECT }
+    );
```

Comment:
Removing `Export.mostRecent` was simple... except for the sweeper.  This Query will keep the the most recent `complete` export per profile.   There's different logic between SQLite and Postgres to know how many Exports were delete.